File: gbdb/src/main/scala/com/graphbrain/gbdb/SearchInterface.scala
package com.graphbrain.gbdb


class SearchInterface(store: VertexStore) {

  def query(text: String) = {
    val id = ID.sanitize(text)
    var maxId = 0

    while (store.exists("" + (maxId + 1) + "/" + id))
      maxId += 1

    for (i <- 1 to maxId) yield "" + i + "/" + id
  }
}

File: graphbrain/db/searchinterface.py
package com.graphbrain.db;

public class SearchInterface {
	
	private Graph graph;
	
	public SearchInterface(Graph graph) {
		this.graph = graph;
	}
	
	public String[] query(String text) {
		String id = ID.sanitize(text);
		int maxId = 0;

		while (graph.exists("" + (maxId + 1) + "/" + id)) {
			maxId += 1;
		}

		String res[] = new String[maxId];
		for (int i = 1; i <= maxId; i++) {
			res[i] = "" + i + "/" + id;
		}
		return res;
	}
}